### ISSUE TITLE
`'var' is declared but its value is never read` false-positive

### ISSUE THREAD

**Issue by boris-petrov** [microsoft/TypeScript#51951]
# Bug Report

### 🔎 Search Terms

`Nullish coalescing assignment`
`declared but its value is never read`

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about Nullish coalescing assignment

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.9.4#code/MYGwhgzhAECC0G8BQ1oAcBOBLAbmALgKbQAeA-AFzQB2ArgLYBGhGA3EiurYyFsNADMA9kIAUASio4hWACaJOqfAAssEAHQloZMgF5oEQvgAqWeoSG18oidF0A+RAF9x7VE6ROgA)

### 💻 Code

```ts
class A {
  private x?: number;

  public foo(): void {
    this.x ??= setTimeout(() => {});
  }
}
```

### 🙁 Actual behavior

There is an error that `'x' is declared but its value is never read` which is not true because `??=` reads and then writes the value.

### 🙂 Expected behavior

No error on `x`.

**Comment by boris-petrov** [comment]
Ah, sorry, you're better at searching for issues than me! :smile:

**Comment by MartinJohns** [comment]
I have many years of practice, and I've seen most reoccurring issues on this repository several times already.
